spec/models/legal_aid_application_transaction_type_spec.rb

```diff
@@ -52,4 +52,49 @@
       expect(debits).to contain_exactly(a_debit)
     end
   end
+
+  describe ".after_commit" do
+    let(:instance) do
+      create(:legal_aid_application_transaction_type,
+             legal_aid_application:,
+             transaction_type: credit_transaction_type)
+    end
+
+    before do
+      create(:legal_aid_application_transaction_type,
+             legal_aid_application:,
+             transaction_type: debit_transaction_type)
+
+      legal_aid_application.cash_transactions.create!(transaction_type_id: credit_transaction_type.id, amount: 101, month_number: 1, transaction_date: Time.zone.now.to_date)
+      legal_aid_application.cash_transactions.create!(transaction_type_id: debit_transaction_type.id, amount: 103, month_number: 3, transaction_date: 2.months.ago)
+
+      allow(instance).to receive(:cascade_delete_cash_transactions).and_call_original
+    end
+
+    context "when destroying object instance" do
+      let(:action) { instance.destroy! }
+
+      it "does not call cascade_delete_cash_transactions" do
+        action
+        expect(instance).to have_received(:cascade_delete_cash_transactions)
+      end
+
+      it "cascades delete to cash transactions of that type" do
+        expect { action }.to change(legal_aid_application.cash_transactions, :count).by(-1)
+      end
```

Comment:
I wonder if we need both these tests. Perhaps we should focus on testing the behaviour, and delete the other test?

Something like this would probably be sufficient

```ruby
describe ".after_commit" do
  context "when destroying the transaction type" do
    it "destroys that type's cash transactions" do
      setup
      
      destroy_the_transaction_type

      assert_the_correct_transactions_are_destroyed
    end
  end 

  context "when updating the transaction type" do
    it "does not destroy that type's cash transactions" do
      setup
      
      destroy the transaction_type
      
      assert_the_transactions_are_not_destroyed
    end
  end
end
```

What do you think?